docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -223,10 +223,79 @@ features, like:
 Note that data modification also requires implementing
 a :ref:`connector-page-sink-provider`.
 
+When Trino receives a ``SELECT`` query, it parses it into an Intermediate
+Representation (IR). Then, during optimization, it checks if connectors
+can handle operations related to SQL clauses by calling one of the following
+methods of the ``ConnectorMetadata`` service:
+
+* ``applyLimit``
+* ``applyTopN``
+* ``applyFilter``
+* ``applyProjection``
+* ``applySample``
+* ``applyAggregation``
+* ``applyJoin``
+* ``applyTableFunction``
+* ``applyTableScanRedirect``
+
+Connectors can indicate that they don't support a particular pushdown or that
+the action had no effect by returning ``Optional.empty()``. Connectors should
+expect these methods to be called multiple times during the optimization of
+a given query.
+
+.. warning::
+
+  It's critical for connectors to return ``Optional.empty()`` if calling
+  this method has no effect for that invocation, even if the connector generally
+  supports a particular pushdown. Doing otherwise can cause the optimizer
+  to loop indefinitely.
+
+Otherwise, these methods return a result object containing the connectors
+table handle. The connector can store any kind of information in the table
+handle, as it's passed to other services it implements, like
+the ``ConnectorRecordSetProvider`` or ``ConnectorPageSourceProvider``.
+
+Limit and top-N pushdown
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+When executing a ``SELECT`` query with ``LIMIT`` or ``ORDER BY`` clauses,
+the query plan may contain a ``Sort`` or ``Limit`` operations.
+
+When the plan contains a ``Sort`` operation, the engine
+tries to push down the limit into the connector by calling the ``applyTopN``
```

Comment:
This is only when there is a Sort *and* a Limit (they first get converted to a TopN IR operation). A Sort alone would not result in a call to applyTopN

Reply:
How does that map to a query like `SELECT * FROM a ORDER BY b`? Would there be a `Limit` operation with a max value or it wouldn't push down the order at all?

Reply:
Order won't be pushed.

Reply:
Thanks. I can't explain why I thought this works differently. I checked the tests in my custom connectors and I couldn't find any with only `ORDER BY`. I guess this is why we need to have this in the developer guide :-)